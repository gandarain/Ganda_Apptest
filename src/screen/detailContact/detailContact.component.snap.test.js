import React from 'react'
import renderer from 'react-test-renderer'

import { mockContact } from '../../fixture'

import DetailContact from './detailContact.component'

jest
  .mock('../../component/header', () => 'Header')
  .mock('../../component/loading', () => 'Loading')
  .mock('../../component/loader', () => 'Loader')
  .mock('react-native-config', () => ({
    BASE_URL: ''
  }))
  .mock('../../service', () => ({
    get: () => Promise.resolve(mockContact)
  }))
  .mock('@react-navigation/native', () => ({
    useNavigation: () => ({
      navigation: () => ({
        navigate: jest.fn()
      })
    })
  }))
  .mock('react-redux', () => ({
    useDispatch: jest.fn()
  }))
  .mock('../../redux/action/listContactsActions', () => ({
    setLoadingListContacts: jest.fn()
  }))

describe('Detail Contact Component', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should render loading', () => {
    const loading = true
    const contact = {}
    const loader = false
    const setStateMock = jest.fn()

    React.useState = jest.fn().mockReturnValue([loader, {}])
    React.useState = jest.fn().mockReturnValue([contact, {}])
    React.useState = jest.fn().mockReturnValue([loading, {}])

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [loading, setStateMock])
      .mockImplementationOnce(() => [contact, setStateMock])
      .mockImplementationOnce(() => [loader, setStateMock])

    const rendered = renderer.create(<DetailContact />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render detail contact component', () => {
    const loading = false
    const contact = mockContact.data.data
    const loader = false
    const setStateMock = jest.fn()

    React.useState = jest.fn().mockReturnValue([loader, {}])
    React.useState = jest.fn().mockReturnValue([contact, {}])
    React.useState = jest.fn().mockReturnValue([loading, {}])

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [loading, setStateMock])
      .mockImplementationOnce(() => [contact, setStateMock])
      .mockImplementationOnce(() => [loader, setStateMock])

    const rendered = renderer.create(<DetailContact />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render loader', () => {
    const loading = false
    const contact = mockContact.data.data
    const loader = true
    const setStateMock = jest.fn()

    React.useState = jest.fn().mockReturnValue([loader, {}])
    React.useState = jest.fn().mockReturnValue([contact, {}])
    React.useState = jest.fn().mockReturnValue([loading, {}])

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [loading, setStateMock])
      .mockImplementationOnce(() => [contact, setStateMock])
      .mockImplementationOnce(() => [loader, setStateMock])

    const rendered = renderer.create(<DetailContact />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
