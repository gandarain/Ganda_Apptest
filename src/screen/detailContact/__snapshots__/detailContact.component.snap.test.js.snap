// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Contact Component should render detail contact component 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={
      [
        {
          "action": [Function],
          "icon": "pencil-outline",
        },
        {
          "action": [Function],
          "icon": "delete-outline",
        },
      ]
    }
    showBackButton={true}
    title="Detail Contact"
  />
  <RCTScrollView
    accessibilityLabel="ContainerDetailContact"
  >
    <View>
      <Image
        accessibilityLabel="ImageDetailContact"
        resizeMode="cover"
        source={
          {
            "uri": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
          }
        }
        style={
          {
            "height": 300,
            "width": "100%",
          }
        }
      />
      <View
        accessibilityLabel="NameDetailContact"
        style={
          {
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "marginHorizontal": 15,
            "marginTop": 15,
            "paddingBottom": 15,
          }
        }
      >
        <Text
          accessibilityLabel="TitleNameDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Bold",
              "fontSize": 18,
              "marginBottom": 10,
            }
          }
        >
          Name
        </Text>
        <Text
          accessibilityLabel="TextNameDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
            }
          }
        >
          Bilbo
           
          Baggins
        </Text>
      </View>
      <View
        accessibilityLabel="AgetDetailContact"
        style={
          {
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "marginHorizontal": 15,
            "marginTop": 15,
            "paddingBottom": 15,
          }
        }
      >
        <Text
          accessibilityLabel="TitleAgeDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Bold",
              "fontSize": 18,
              "marginBottom": 10,
            }
          }
        >
          Age
        </Text>
        <Text
          accessibilityLabel="TextAgeDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
            }
          }
        >
          111
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <Loader
    onClose={[Function]}
    visible={false}
  />
</View>
`;

exports[`Detail Contact Component should render loader 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={
      [
        {
          "action": [Function],
          "icon": "pencil-outline",
        },
        {
          "action": [Function],
          "icon": "delete-outline",
        },
      ]
    }
    showBackButton={true}
    title="Detail Contact"
  />
  <RCTScrollView
    accessibilityLabel="ContainerDetailContact"
  >
    <View>
      <Image
        accessibilityLabel="ImageDetailContact"
        resizeMode="cover"
        source={
          {
            "uri": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
          }
        }
        style={
          {
            "height": 300,
            "width": "100%",
          }
        }
      />
      <View
        accessibilityLabel="NameDetailContact"
        style={
          {
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "marginHorizontal": 15,
            "marginTop": 15,
            "paddingBottom": 15,
          }
        }
      >
        <Text
          accessibilityLabel="TitleNameDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Bold",
              "fontSize": 18,
              "marginBottom": 10,
            }
          }
        >
          Name
        </Text>
        <Text
          accessibilityLabel="TextNameDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
            }
          }
        >
          Bilbo
           
          Baggins
        </Text>
      </View>
      <View
        accessibilityLabel="AgetDetailContact"
        style={
          {
            "borderBottomColor": "#EEEEEE",
            "borderBottomWidth": 1,
            "marginHorizontal": 15,
            "marginTop": 15,
            "paddingBottom": 15,
          }
        }
      >
        <Text
          accessibilityLabel="TitleAgeDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Bold",
              "fontSize": 18,
              "marginBottom": 10,
            }
          }
        >
          Age
        </Text>
        <Text
          accessibilityLabel="TextAgeDetailContact"
          style={
            {
              "color": "#5E5E5E",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
            }
          }
        >
          111
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <Loader
    onClose={[Function]}
    visible={true}
  />
</View>
`;

exports[`Detail Contact Component should render loading 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={
      [
        {
          "action": [Function],
          "icon": "pencil-outline",
        },
        {
          "action": [Function],
          "icon": "delete-outline",
        },
      ]
    }
    showBackButton={true}
    title="Detail Contact"
  />
  <Loading />
  <Loader
    onClose={[Function]}
    visible={false}
  />
</View>
`;
